app/controllers/admin/base_controller.rb

```diff
@@ -6,13 +6,25 @@ class BaseController < ApplicationController
 
     include FakeExceptionHelper
 
-    if Rails.env.production?
-      before_filter :authenticate_admin!
-    else
+    if Rails.env.development?
       skip_before_filter :authenticate_user!
       skip_before_filter :finish_sign_up
 
       fine_print_skip :general_terms_of_use, :privacy_policy
+    else
+      before_filter :authenticate_admin!
+      before_filter :log_out_inactive_admins
+    end
+
+    def log_out_inactive_admins
+      if current_user.is_administrator?
+        if session[:last_admin_activity].to_time <= 30.minutes.ago
+          sign_out!
+          authenticate_admin!
```

Comment:
I agree, the first authenticate_admin! call is not necessary, but if I do remove the first one then I should rename log_out_inactive_admins to something more descriptive such like authenticate_active_admin.